Finance unit testing failing in Travis build

https://github.com/ericmjl/pyjanitor/pull/146#issuecomment-482982960

Test log:

```
=================================== FAILURES ===================================
________________________ test_make_currency_api_request ________________________
    @pytest.mark.finance
    def test_make_currency_api_request():
>       r = requests.get("https://api.exchangeratesapi.io")
tests/finance/test_convert_currency.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/api.py:75: in get
    return request('get', url, params=params, **kwargs)
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/api.py:60: in request
    return session.request(method=method, url=url, **kwargs)
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/sessions.py:533: in request
    resp = self.send(prep, **send_kwargs)
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/sessions.py:668: in send
    history = [resp for resp in gen] if allow_redirects else []
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/sessions.py:668: in <listcomp>
    history = [resp for resp in gen] if allow_redirects else []
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <requests.sessions.Session object at 0x7f9d295a5eb8>
resp = <Response [301]>, req = <PreparedRequest [GET]>, stream = False
timeout = None, verify = True, cert = None
proxies = OrderedDict([('travis_apt', 'http://build-cache.travisci.net')])
yield_requests = False, adapter_kwargs = {}
hist = [<Response [301]>, <Response [301]>, <Response [301]>, <Response [301]>, <Response [301]>, <Response [301]>, ...]
url = 'https://api.exchangeratesapi.io/', previous_fragment = ''
prepared_request = <PreparedRequest [GET]>
parsed = ParseResult(scheme='https', netloc='api.exchangeratesapi.io', path='/', params='', query='', fragment='')
purged_headers = ('Content-Length', 'Content-Type', 'Transfer-Encoding')
    def resolve_redirects(self, resp, req, stream=False, timeout=None,
                          verify=True, cert=None, proxies=None, yield_requests=False, **adapter_kwargs):
        """Receives a Response. Returns a generator of Responses or Requests."""
    
        hist = []  # keep track of history
    
        url = self.get_redirect_target(resp)
        previous_fragment = urlparse(req.url).fragment
        while url:
            prepared_request = req.copy()
    
            # Update history and keep track of redirects.
            # resp.history must ignore the original request in this loop
            hist.append(resp)
            resp.history = hist[1:]
    
            try:
                resp.content  # Consume socket so it can be released
            except (ChunkedEncodingError, ContentDecodingError, RuntimeError):
                resp.raw.read(decode_content=False)
    
            if len(resp.history) >= self.max_redirects:
>               raise TooManyRedirects('Exceeded %s redirects.' % self.max_redirects, response=resp)
E               requests.exceptions.TooManyRedirects: Exceeded 30 redirects.
../../../miniconda/envs/pyjanitor-dev/lib/python3.7/site-packages/requests/sessions.py:165: TooManyRedirects
```
https://travis-ci.org/ericmjl/pyjanitor/jobs/519783548

--- a/janitor/biology.py
+++ b/janitor/biology.py
@@ -3,7 +3,6 @@ Biology and bioinformatics-oriented data cleaning functions.
 """
 
 import pandas as pd
-
 import pandas_flavor as pf
 
 from .utils import import_message

--- a/janitor/chemistry.py
+++ b/janitor/chemistry.py
@@ -6,7 +6,6 @@ from typing import Union
 
 import numpy as np
 import pandas as pd
-
 import pandas_flavor as pf
 
 from .utils import import_message

--- a/janitor/finance.py
+++ b/janitor/finance.py
@@ -6,9 +6,9 @@ import json
 from datetime import date, datetime
 from functools import lru_cache
 
+import pandas_flavor as pf
 import requests
 
-import pandas_flavor as pf
 from janitor import check
 
 currency_set = {

--- a/janitor/functions.py
+++ b/janitor/functions.py
@@ -10,10 +10,9 @@ from typing import Dict, Iterable, List, Union
 
 import numpy as np
 import pandas as pd
+import pandas_flavor as pf
 from scipy.stats import mode
 from sklearn.preprocessing import LabelEncoder
-
-import pandas_flavor as pf
 
 from .errors import JanitorError
 

--- a/janitor/testing_utils/strategies.py
+++ b/janitor/testing_utils/strategies.py
@@ -3,7 +3,6 @@ Testing strategies are placed here.
 """
 
 import numpy as np
-
 from hypothesis import strategies as st
 from hypothesis.extra.pandas import column, data_frames, range_indexes
 

--- a/tests/finance/test_convert_currency.py
+++ b/tests/finance/test_convert_currency.py
@@ -7,6 +7,17 @@ import janitor.finance
 @pytest.mark.finance
 @pytest.mark.xfail
 def test_make_currency_api_request():
+    """
+    Test for currency API request.
+
+    This test exists because we rely solely on the service by
+    exchangeratesapi. That said, we also mark it as expected to fail because
+    it sometimes pings the exchange rates API a too frequently and causes
+    tests to fail.
+
+    For an example of how this test fails, see:
+    https://github.com/ericmjl/pyjanitor/issues/147
+    """
     r = requests.get("https://api.exchangeratesapi.io")
     assert r.status_code == 200
 

--- a/tests/finance/test_convert_currency.py
+++ b/tests/finance/test_convert_currency.py
@@ -5,6 +5,7 @@ import janitor.finance
 
 
 @pytest.mark.finance
+@pytest.mark.xfail
 def test_make_currency_api_request():
     r = requests.get("https://api.exchangeratesapi.io")
     assert r.status_code == 200

--- a/tests/functions/test_add_column.py
+++ b/tests/functions/test_add_column.py
@@ -1,9 +1,9 @@
 import numpy as np
 import pandas as pd
 import pytest
-
 from hypothesis import given
 from hypothesis import strategies as st
+
 from janitor.testing_utils.strategies import df_strategy
 
 

--- a/tests/functions/test_add_columns.py
+++ b/tests/functions/test_add_columns.py
@@ -1,10 +1,10 @@
 import numpy as np
 import pandas as pd
 import pytest
-
 from hypothesis import assume, given
 from hypothesis import strategies as st
 from hypothesis.extra.numpy import arrays
+
 from janitor.testing_utils.strategies import df_strategy
 
 

--- a/tests/functions/test_clean_names.py
+++ b/tests/functions/test_clean_names.py
@@ -1,7 +1,7 @@
 import pandas as pd
 import pytest
-
 from hypothesis import given
+
 from janitor.errors import JanitorError
 from janitor.testing_utils.strategies import df_strategy
 

--- a/tests/functions/test_encode_categorical.py
+++ b/tests/functions/test_encode_categorical.py
@@ -1,6 +1,6 @@
 import pytest
-
 from hypothesis import given
+
 from janitor.errors import JanitorError
 from janitor.testing_utils.strategies import (
     categoricaldf_strategy,

--- a/tests/functions/test_filter_column_isin.py
+++ b/tests/functions/test_filter_column_isin.py
@@ -1,7 +1,7 @@
 import pandas as pd
 import pytest
-
 from hypothesis import assume, given
+
 from janitor.testing_utils.strategies import (
     categoricaldf_strategy,
     names_strategy,

--- a/tests/functions/test_get_features_targets.py
+++ b/tests/functions/test_get_features_targets.py
@@ -1,6 +1,6 @@
 import pytest
-
 from hypothesis import given
+
 from janitor.testing_utils.strategies import df_strategy
 
 

--- a/tests/functions/test_remove_empty.py
+++ b/tests/functions/test_remove_empty.py
@@ -1,7 +1,7 @@
 import pandas as pd
 import pytest
-
 from hypothesis import given
+
 from janitor.testing_utils.strategies import df_strategy
 
 

--- a/tests/functions/test_rename_column.py
+++ b/tests/functions/test_rename_column.py
@@ -1,5 +1,4 @@
 import pytest
-
 from hypothesis import given
 
 

--- a/tests/functions/test_reorder_columns.py
+++ b/tests/functions/test_reorder_columns.py
@@ -1,6 +1,6 @@
 import pytest
-
 from hypothesis import given
+
 from janitor.testing_utils.strategies import df_strategy
 
 

